test/e2e/flowexporter_test.go

```diff
@@ -0,0 +1,135 @@
+	}
+
+	// Parse through IPFIX collector output
+	re := regexp.MustCompile("(?m)^.*" + "#" + ".*$[\r\n]+")
+	collectorOutput = re.ReplaceAllString(collectorOutput, "")
+	collectorOutput = strings.TrimSpace(collectorOutput)
+	recordSlices := strings.Split(collectorOutput, "IPFIX-HDR:")
+	// Delete the first element from recordSlices
+	recordSlices[0] = recordSlices[len(recordSlices)-1]
+	recordSlices[len(recordSlices)-1] = ""
+	recordSlices = recordSlices[:len(recordSlices)-1]
+	// Iterate over recordSlices and build some results to test with expected results
+	templateRecords := 0
+	dataRecordsIntraNode := 0
+	for _, record := range recordSlices {
+		if strings.Contains(record, "TEMPLATE RECORD") {
+			templateRecords = templateRecords + 1
+		}
+
+		if strings.Contains(record, podAIP) && strings.Contains(record, podBIP) {
+			dataRecordsIntraNode = dataRecordsIntraNode + 1
+			// Check if records have both Pod name and Pod namespace or not
+			if !strings.Contains(record, hex.EncodeToString([]byte("perftest-a"))) {
+				t.Fatalf("Records with podAIP does not have pod name")
+			}
+			if !strings.Contains(record, hex.EncodeToString([]byte("perftest-b"))) {
+				t.Fatalf("Records with podBIP does not have pod name")
+			}
+			if !strings.Contains(record, hex.EncodeToString([]byte(testNamespace))) {
+				t.Fatalf("Records with podAIP and podBIP does not have pod namespace")
+			}
+			// Check the bandwidth using octetDeltaCount in data records sent in second ipfix interval
+			if strings.Contains(record, "seqno=2") || strings.Contains(record, "seqno=3") {
+				// One of them has no bytes ignore that
+				if !strings.Contains(record, "octetDeltaCount: 0") {
+					//split the record in lines to compute bandwidth
+					splitLines := strings.Split(record, "\n")
+					for _, line := range splitLines {
+						if strings.Contains(line, "octetDeltaCount") {
+							lineSlice := strings.Split(line, ":")
+							deltaBytes, err := strconv.ParseFloat(strings.TrimSpace(lineSlice[1]), 64)
+							if err != nil {
+								t.Fatalf("Error in converting octetDeltaCount to int type")
+							}
+							// compute the bandwidth using 5s as interval
+							recBandwidth := (deltaBytes * 8.0) / (5.0 * math.Pow10(9))
+							// bandwidth from iperf output
+							bwSlice := strings.Split(bandwidth, " ")
+							iperfBandwidth, err := strconv.ParseFloat(bwSlice[0], 64)
+							if err != nil {
+								t.Fatalf("Error in converting iperf bandwidth to float64 type")
+							}
+							// Check if at least the first digit is equal, i.e., 42 Gb/s and 48 Gb/s are considered equal
+							// we cannot guarantee both will be exactly same. Logging both values to give visibility.
+							t.Logf("Iperf bandwidth: %v", iperfBandwidth)
+							t.Logf("IPFIX record bandwidth: %v", recBandwidth)
+							assert.Equal(t, int(recBandwidth/10), int(float64(iperfBandwidth)/10), "Iperf bandwidth and IPFIX record bandwidth should be similar")
```

Comment:
that sounds like a recipe for flakiness: what about 39Gbps and 41Gbps?

please consider using something like https://godoc.org/github.com/stretchr/testify/assert#InEpsilon

Reply:
done